public/editor-client/src/types/DefaultTemplate.ts

```diff
@@ -215,3 +204,66 @@ export interface PopupsWithThumbs extends Omit<Popups, "blocks"> {
   blocks: Array<BlockWithThumbs>;
 }
 // endregion
+
+// region Layouts
+export interface LayoutsDefaultTemplate<T1, T2, T3> {
+  label?: string;
+  getMeta: (res: Response<T1>, rej: Response<string>) => void;
+  getData: (
+    res: Response<T2>,
+    rej: Response<string>,
+    page: CustomTemplatePage
+  ) => void;
+  getPages: (res: Response<T3>, rej: Response<string>, id: string) => void;
+}
+
+export interface LayoutTemplate {
+  blank?: boolean;
+  name: string;
+  cat: Array<number | string>;
+  pagesCount: number;
+  layoutId: string;
+  pro: boolean;
+  keywords: string;
+}
+
+export interface Layouts {
+  templates: Array<LayoutTemplateWithThumbs>;
+  categories: Pick<Categories, "id" | "title">[];
+}
+
+export interface LayoutsPages {
+  pages: CustomTemplatePage[];
+  styles: Style[];
+}
+
+export interface LayoutsWithThumbs extends Omit<Layouts, "templates"> {
+  templates: Array<LayoutTemplateWithThumbs>;
+}
+
+export interface LayoutTemplateWithThumbs extends LayoutTemplate {
+  thumbnailSrc: string;
+  thumbnailWidth: number;
+  thumbnailHeight: number;
+}
+
+export type LayoutsPageAPI = {
+  title: string;
+  slug: string;
+  thumbs: string;
+  thumbnailWidth: number;
+  thumbnailHeight: number;
+};
+
+export type LayoutsAPI = {
+  title: string;
+  pro: string;
+  categories: string;
+  pagesCount: string;
+  slug: string;
+  thumbnail: string;
+  thumbnailWidth: number;
+  thumbnailHeight: number;
```

Comment:
create an interface with common properties and use as extend for `LayoutsPageAPI` and `LayoutsAPI`, as example:
```ts
interface Common {
  title: string;
  slug: string;
  /* other */
}

interface LayoutsPageAPI extends Common {}
interface LayoutsAPI extends Common {}

```